Build fails for `stdlib` with `jsonv2` experiment enabled

If `jsonv2` experiment is enabled building of stdlib fails with the following error:
```
Stdlib command (from target @@rules_go+//:stdlib) bazel-out/darwin_arm64-opt-exec-ST-8d526891f1c5/bin/external/rules_go++go_sdk+go_sdk/builder_reset/builder stdlib -sdk external/rules_go++go_sdk+go_sdk -goroot external/rules_go++go_sdk+go_sdk ... (remaining 10 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
bazel-out/darwin_arm64-fastbuild-ST-629329df9ef8/bin/external/rules_go+/stdlib_/src/encoding/json/internal/jsontest/testdata.go:28:12: pattern testdata/*.json.zst: no matching files found
stdlib: error running subcommand external/rules_go++go_sdk+go_sdk/bin/go: exit status 1
```

Configuration:
```
go_version = "1.25.0"

go_sdk.download(
    name = "go_sdk",
    experiments = [
        "jsonv2",
    ],
    sdks = {
        "darwin_arm64": (
            "go{}.darwin-arm64.tar.gz".format(go_version),
            "544932844156d8172f7a28f77f2ac9c15a23046698b6243f633b0a0b00c0749c",
        ),
        "linux_amd64": (
            "go{}.linux-amd64.tar.gz".format(go_version),
            "2852af0cb20a13139b3448992e69b868e50ed0f8a1e5940ee1de9e19a123b613",
        ),
        "linux_arm64": (
            "go{}.linux-arm64.tar.gz".format(go_version),
            "05de75d6994a2783699815ee553bd5a9327d8b79991de36e38b66862782f54ae",
        ),
    },
)
```

If I attempt to replicate what rules_go does, I run into this panic in native Go tooling:
```
$ GODEBUG=installgoroot=all GOEXPERIMENT=jsonv2 go install std
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x3 pc=0x102546d60]

goroutine 1 [running]:
cmd/go/internal/fips140.Init()
	cmd/go/internal/fips140/fips140.go:117 +0x120
cmd/go/internal/modload.Init()
	cmd/go/internal/modload/init.go:419 +0x38
cmd/go/internal/toolchain.Exec({0x140000ba6ca, 0x8})
	cmd/go/internal/toolchain/select.go:356 +0x46c
cmd/go/internal/toolchain.Select()
	cmd/go/internal/toolchain/select.go:284 +0xf28
main.main()
	cmd/go/main.go:107 +0x58
```

--- a/go/private/BUILD.sdk.bazel
+++ b/go/private/BUILD.sdk.bazel
@@ -25,6 +25,7 @@ filegroup(
     name = "srcs",
     srcs = glob(
         ["src/**/*"],
+        allow_empty = True,
         exclude = [
             "src/**/*_test.go",
             "src/**/testdata/**",

